src/modules/Thread/context/hooks/useHandlePubsubEvents.ts

```diff
@@ -31,30 +32,39 @@ export default function useHandlePubsubEvents({
       if (!pubSub || !pubSub.subscribe) {
         return subscriber;
       }
+      // TODO: subscribe ON_FILE_INFO_UPLOADED
       subscriber.set(topics.SEND_MESSAGE_START, pubSub.subscribe(topics.SEND_MESSAGE_START, (props) => {
-        const { channel, message } = props;
+        const {
+          channel,
+          message,
+          publishingModules,
+        } = props as { channel: GroupChannel, message: SendableMessageType, publishingModules: PublishingModuleType[] };
         if (currentChannel?.url === channel?.url
           && message?.parentMessageId === parentMessage?.messageId
+          && publishingModules.includes(PublishingModuleType.THREAD)
         ) {
-          if (isMultipleFilesMessage(message)) {
-            threadDispatcher({
-              type: ThreadContextActionTypes.SEND_MESSAGE_START,
-              payload: {
-                message: {
-                  ...message,
-                  fileInfoList: message.messageParams.fileInfoList.map((fileInfo) => ({
-                    ...fileInfo,
-                    url: URL.createObjectURL(fileInfo.file as File),
-                  })),
-                },
-              },
-            });
+          // TODO: const clonedMessage = cloneMessage(message);
+          const nextMessage: Record<string, any> = { ...message };
```

Comment:
If possible, let's avoid copying message instance as an object. It loses crucial methods like message.isMultipleFilesMessage() 
cc. @bang9 This is what i was talking about

Reply:
also its minor but i would rename nextMessage to pendingMessage.

Reply:
I made a util function, but we need to take some time to add tests for it.